### ISSUE TITLE
Possibility to locate "dropped" wyvern eggs?

### ISSUE THREAD

**Issue by mitfry** [Qowyn/ark-tools#24]
Hello!

For over a year now certain wyvern nests have been bugged on our Scorched Earth server because of players dropping the wyvern eggs they looted in mid air. The eggs won't de-spawn by themselves and so the nests won't get new eggs. 

The only solution is to pick-up the eggs that are scattered across the map. Insanely difficult to find. I think it would be possible to locate them using the ARK-tools, but I haven't been successful yet. 

Can anybody come up with a command that would export the location of dropped wyvern eggs?

I would like to thank you in advance.

Mitfrey

**Comment by Qowyn** [comment]
Good idea, currently there is no easy way to get a list of eggs "in the world".

**Comment by mitfry** [comment]
I hoped it would be as easy as adding the right info to the ark_data.json file, but I tried that and it didn't work. Don't think I have enough knowledge yet about the save files and ark-tools to figure this one out myself =/.

I found the following matching .uasset files in the ARK content folder:
Location: Content\PrimalEarth\Test
- PrimalItemConsumable_Egg_Wyvern
- PrimalItemConsumable_Egg_Wyvern_Fertilized
- PrimalItemConsumable_Egg_Wyvern_Fertilized_Fire
- PrimalItemConsumable_Egg_Wyvern_Fertilized_Lightning
- PrimalItemConsumable_Egg_Wyvern_Fertilized_Poison

Location: Content\ScorchedEarth\Dinos\Wyvern\Egg
- dragon_egg_1
- dragon_egg_2
- dragon_egg_2
- DragonEgg1_MIC
- DragonEgg2_MIC
- DragonEgg3_MIC

I'm not even sure these files would be remotely useful in finding info about the eggs in the save file, but it might save you some searching. 

You've got any ideas about possible approaches to locate this data?

**Comment by Qowyn** [comment]
It's now available by using the `tribes` command with the new `--non-players` switch together with `--items long`. This will get you a listing of all items not belonging to any player, including unfertilized eggs, as a short-term solution. For the long-term i want to add a query command, making it possible to query the map-save for vitually anything.

**Comment by mitfry** [comment]
Oh WOW! If this works, I'm gonna go crazy! 

I'll try right now, reporting back soon!

**Comment by mitfry** [comment]
Alright, so I was able to successfully run the command and get the following output file:
[non-players.json.txt](https://github.com/Qowyn/ark-tools/files/1306754/non-players.json.txt)

This is the command I used:
ark-tools.exe tribes --pretty .\SavedArks\ScorchedEarth_P.ark .\Ark-Tools --non-player --items long

However, unless I'm missing something or used the command in the wrong way, I did not find the locations info of the items in the output file.

**Comment by Qowyn** [comment]
Damn... right, it's just giving you the items themselves, without the containing "Object". Items in ARK aren't "real" objects, causing them to have no position.

**Comment by mitfry** [comment]
We'll get there! The location data of these floating eggs must be saved somewhere.. 

What can I do to help you?

**Comment by Qowyn** [comment]
Added support for full dropped item reading, use with `--inventory long` instead of `--items long`

**Comment by mitfry** [comment]
Oh WOW! It seems to be working! Below is my output file!
[non-players.json.txt](https://github.com/Qowyn/ark-tools/files/1307493/non-players.json.txt)

It's been over a full year for the devs to simply implement a fix to have the eggs despawn so the nests keep working. Cannot wait to finally clean this up.  I'm gonna check right now whether these are the missing eggs.

Thanks a LOT for your efforts so far Qowyn!
